vortex-scalar/src/struct_.rs

```diff
@@ -57,19 +57,23 @@ impl<'a> StructScalar<'a> {
             })
     }
 
-    pub fn field(&self, name: &str) -> Option<Scalar> {
+    pub fn maybe_null_field(&self, name: &str) -> Option<Scalar> {
         let DType::Struct(st, _) = self.dtype() else {
             unreachable!()
         };
-        st.find_name(name).and_then(|idx| self.field_by_idx(idx))
+        st.find_name(name)
+            .and_then(|idx| self.maybe_null_field_by_idx(idx))
     }
 
     pub fn fields(&self) -> Option<Vec<Scalar>> {
```

Comment:
This naming seems inconsistent, imho there's no problem with the current `field() -> Option`

Reply:
it ignores struct level validity, see https://github.com/spiraldb/vortex/issues/1513. @danking is working on a compute function which will allow `field` which applies the validity of the struct to its children.

Reply:
Where would you say this naming is inconsistent with?

Reply:
`fields` and `field` seem consistent. I don't think we need `maybe_null`, the `Option` return type expresses that.

Reply:
That is the point, `Option` represents that the field-idx might not be in the struct array. Maybe null means that some elements of the returned array are null

Reply:
Scalars don't suffer from the same issue as the array. If top level validity is invalid the scalar is null. But also we went back on `maybe_null_slice` and just have `as_slice` now. I think shorter names are better assuming the contract of the api is documented

Reply:
in fact i see this is on scalars, maybe I am wrong

Reply:
indeed StructArray#scalar_at ignores validity which I think is a bug that should be fixed

Reply:
Although its does seem strange that the Option is either idx out of range or value is nullable? I wonder if the out of range should be a panic or if this should result<option<scalar>>?

Reply:
@robert3005 When you project out a field, there is not currently not a way to add a validity mask to that, so scalar_at cannot know

Reply:
Semantics here are weird. I would guess that null value would still produce a null field. Even though the value is null we still know the schema but we use same value to convey both information

Reply:
I have updated the struct scalar